### ISSUE TITLE
Flutter run is building, but not updating the running code, when using --profile

### ISSUE THREAD

**Issue by gspencergoog** [flutter/flutter#17155]
## Steps to Reproduce

1. Create a new project with `flutter create no_rebuild`
1. Run the app with `flutter run --profile`
1. Quit the app with `q` on the command line
1. Edit the app's main.dart and comment out the FAB button.
1. Run the app with `flutter run --profile`
1. Notice that the FAB is gone.
1. Quit the app again
1. Edit the app's main.dart and uncomment the FAB button you commented out.
1. Run the app with `flutter run --profile`
1. Notice that the FAB does **NOT** reappear as it should

I've seen this a lot, and after that point, the only way I can get it to reload the new code for the app is to run `flutter clean` to remove the build directory.

It seems to get to a point where it never updates again after one update.

## Flutter Doctor

```
[✓] Flutter (Channel chip_perf, v0.3.6-pre.30, on Mac OS X 10.13.3 17D102, locale en-US)
    • Flutter version 0.3.6-pre.30 at /Users/gspencer/code/flutter
    • Framework revision 5beb182ad1 (28 minutes ago), 2018-05-01 08:22:29 -0700
    • Engine revision 613aae032f
    • Dart version 2.0.0-dev.50.0.flutter-cd0c4e4fe1

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.2)
    • Android SDK at /Users/gspencer/Library/Android/sdk
    • Android NDK at /Users/gspencer/Library/Android/sdk/ndk-bundle
    • Platform android-27, build-tools 26.0.2
    • ANDROID_HOME = /Users/gspencer/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    • ios-deploy 1.9.2
    • CocoaPods version 1.4.0

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    • Dart plugin version 162.2924
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] IntelliJ IDEA Community Edition (version 2018.1.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 23.2.3
    • Dart plugin version 181.4445.29

[✓] Connected devices (1 available)
    • Pixel 2 • FA81A1A02109 • android-arm64 • Android 8.1.0 (API 27)

• No issues found!
```

**Comment by gspencergoog** [comment]
Best solution right now is to run `flutter clean` between each run.  Yeah, it's slow. @cbracken  is going to take a look.

**Comment by cbracken** [comment]
This should be fixed by #17175. Please shout if you continue to see this.

**Comment by thevikke** [comment]
Can confirm!

**Comment by zoechi** [comment]
This bug is long closed. Please create a new issue.
Perhaps try `flutter channel `dev` first. I think there were some related changes recently and the default channel `beta` might not yet contain these changes.

**Comment by manojvirat457** [comment]
I am also getting the same issue in android studio. The code is not updating on the installation. After a hot reload, it is reflected

**Comment by Devizz998** [comment]
Same problem here....

**Comment by pearlmaknun** [comment]
same here

**Comment by markbruk** [comment]
Same problem here but only after I installed VS Code.  Now I have to Flutter Hot Restart after changing a letter in a text displayed word.  Before installing VS Code as soon as I hit SAVE, I saw the change in my emulator.

**Comment by cbracken** [comment]
As noted, this issue was closed two years ago. 

Please file a new issue with detailed repro steps if you’re seeing this today. It’s possible that changes have landed in the 2 years since this was fixed that triggered a similar regression.
